packages/nouns-webapp/src/wrappers/nounsDao.ts

```diff
@@ -646,3 +929,430 @@ export const useExecuteProposal = () => {
+      method: 'GRACE_PERIOD',
+      args: [],
+    }) || [];
+
+  return gracePeriod?.toNumber();
+};
+
+export const useAdjustedTotalSupply = (): number | undefined => {
+  const [totalSupply] =
+    useContractCall<[EthersBN]>({
+      abi,
+      address: nounsDaoContract.address,
+      method: 'adjustedTotalSupply',
+    }) || [];
+  return totalSupply?.toNumber();
+};
+
+export const useForkThresholdBPS = (): number | undefined => {
+  const [forkThresholdBPS] =
+    useContractCall<[EthersBN]>({
+      abi,
+      address: nounsDaoContract.address,
+      method: 'forkThresholdBPS',
+    }) || [];
+  return forkThresholdBPS?.toNumber();
+};
+
+export const useActivePendingUpdatableProposers = (blockNumber: number) => {
+  const {
+    loading,
+    data: proposals,
+    error,
+  } = useQuery(activePendingUpdatableProposersQuery(1000, blockNumber)) as {
+    loading: boolean;
+    data: { proposals: ProposalProposerAndSigners[] };
+    error: Error;
+  };
+  let data: string[] = [];
+  proposals?.proposals.length > 0 &&
+    proposals.proposals.map(proposal => {
+      data.push(proposal.proposer.id);
+      proposal.signers.map((signer: { id: string }) => {
+        data.push(signer.id);
+        return signer.id;
+      });
+      return proposal.proposer.id;
+    });
+
+  return {
+    loading,
+    data,
+    error,
+  };
+};
+
+export const useHasActiveOrPendingProposalOrCandidate = (account: string, timestamp: number) => {
+  const { data } = useQuery(activePendingUpdatableProposersQuery(1000, timestamp));
+  // check to see if returned proposals have a proposer that matches the account
+  const hasActiveOrPendingProposalOrCandidate = data?.proposals?.some(
+    (proposal: ProposalSubgraphEntity) => {
+      if (proposal.proposer.id.toLowerCase() === account.toLowerCase()) {
+        return true;
+      }
+      if (
+        proposal.signers.some(
+          (signer: { id: string }) => signer.id.toLowerCase() === account.toLowerCase(),
+        )
+      ) {
+        return true;
+      }
+      return false;
+    },
+  );
+
+  return hasActiveOrPendingProposalOrCandidate;
+};
+
+export const checkHasActiveOrPendingProposalOrCandidate = (
+  latestProposalStatus: ProposalState,
+  latestProposalProposer: string | undefined,
+  account: string | null | undefined,
+) => {
+  if (
+    account &&
+    latestProposalProposer &&
+    (latestProposalStatus === ProposalState.ACTIVE ||
+      latestProposalStatus === ProposalState.PENDING ||
+      latestProposalStatus === ProposalState.UPDATABLE) &&
+    latestProposalProposer.toLowerCase() === account?.toLowerCase()
+  ) {
+    return true;
+  }
+  return false;
+};
+
+export const useIsDaoGteV3 = (): boolean => {
+  const [implementation] =
+    useContractCall({
+      abi: abiV2,
+      address: config.addresses.nounsDAOProxy,
+      method: 'implementation',
+    }) || [];
+  if (
+    implementation &&
+    config.addresses.nounsDAOLogicV2 &&
```

Comment:
@ripe0x heads up, this address in `addresses.json` is NOT the latest logic we're using on mainnet is `0x51C7D7C47E440d937208bD987140D6db6B1E4051`